sonar-kotlin-checks/src/main/java/org/sonarsource/kotlin/checks/ServerCertificateCheck.kt

```diff
@@ -104,6 +125,43 @@ class ServerCertificateCheck : AbstractCheck() {
         }
     }
 
+    private class AndroidWebViewCandidateVisitor : KtVisitorVoid() {
+        private val androidSslErrorHandlerProceedFunMatcher = FunMatcher {
+            definingSupertype = "android.webkit.SslErrorHandler"
+            withNames("proceed")
+        }
+
+        private val androidSslErrorHandlerCancelFunMatcher = FunMatcher {
+            definingSupertype = "android.webkit.SslErrorHandler"
+            withNames("cancel")
+        }
+
+        var foundProceedCall: Boolean = false
+        var foundCancelCall: Boolean = false
+        var potentialBranching: Boolean = false
+
+        override fun visitCallExpression(expression: KtCallExpression) {
+            val isProceedCall = androidSslErrorHandlerProceedFunMatcher.matches(expression)
+            val isCancelCall = androidSslErrorHandlerCancelFunMatcher.matches(expression)
+            foundProceedCall = foundProceedCall || isProceedCall
+            foundCancelCall = foundCancelCall || isCancelCall
+        }
+
+        override fun visitExpression(expression: KtExpression) {
+            potentialBranching = potentialBranching ||
```

Comment:
Educational question: does this cover all branching?

Reply:
Yeah, good question - the answer is: all the ones I could think of :-)
I find unlikely that somebody will find a way to have branching, have `proceed` and not have any cancel, in other ways... but I would not underestimate people creativity :-)
Anyway, unless I am missing some common scenarios, the risk for FP seems close to zero to me. 
If we are concerned about this heuristic we can always remove it go for the Option 3 we discussed (only simple `proceed` in body), but that seems too much on the side of safety, at the expense of reach.
WDYT?